src/odemis/gui/plugin/__init__.py

```diff
@@ -420,10 +423,9 @@ def addStream(self, stream):
         returns (StreamController): the stream entry
 
         """
-
```

Comment:
Please extend addStream() to support passing an index corresponding to the viewport. Similar to the overriding method you've written.